tests/qchem/of_tests/test_molecular_hamiltonian.py

```diff
@@ -225,43 +216,31 @@ def test_differentiable_hamiltonian(symbols, geometry, h_ref_data, op_arithmetic
     geometry.requires_grad = False
     h_noargs = qchem.molecular_hamiltonian(symbols, geometry, method="dhf")[0]
 
-    h_ref = (
-        dot(h_ref_data[0], h_ref_data[1], pauli=True)
-        if op_arithmetic
-        else Hamiltonian(h_ref_data[0], h_ref_data[1])
-    )
-
-    if op_arithmetic:
-        disable_new_opmath()
-        h_args_ps = pauli_sentence(h_args)
-        h_noargs_ps = pauli_sentence(h_noargs)
-        h_ref_sorted_coeffs = np.sort(list(h_ref.values()))
-
-        assert set(h_args_ps) == set(h_ref)
-        assert set(h_noargs_ps) == set(h_ref)
-
-        assert np.allclose(np.sort(list(h_args_ps.values())), h_ref_sorted_coeffs)
-        assert np.allclose(np.sort(list(h_noargs_ps.values())), h_ref_sorted_coeffs)
+    ops = [
+        qml.operation.Tensor(*op) if isinstance(op, qml.ops.Prod) else op
+        for op in map(qml.simplify, h_ref_data[1])
+    ]
+    h_ref = qml.Hamiltonian(h_ref_data[0], ops)
```

Comment:
`qml.Hamiltonian` is called with both legacy and new opmath?

Reply:
Yeah thus would simply return LinearCombination under new opmath.